pkg/operator/ceph/csi/spec.go

```diff
@@ -311,5 +315,45 @@ func StartCSIDrivers(namespace string, clientset kubernetes.Interface, ver *vers
 			return fmt.Errorf("failed to create rbd service: %+v\n%+v", err, cephfsService)
 		}
 	}
+
+	if ver.Minor >= provDeploymentSuppVersion {
+		err = createCSIDriverInfo(clientset, RBDDriverName)
+		if err != nil {
+			return fmt.Errorf("failed to create CSI driver object for %q: %+v", RBDDriverName, err)
+		}
+		err = createCSIDriverInfo(clientset, CephFSDriverName)
+		if err != nil {
+			return fmt.Errorf("failed to create CSI driver object for %q: %+v", CephFSDriverName, err)
+		}
+	}
 	return nil
 }
+
+// createCSIDriverInfo Registers CSI driver by creating a CSIDriver object
+func createCSIDriverInfo(clientset kubernetes.Interface, name string) error {
+	attach := false
+	mountInfo := false
+	// Create CSIDriver object
+	csiDriver := &k8scsi.CSIDriver{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: k8scsi.CSIDriverSpec{
+			AttachRequired: &attach,
+			PodInfoOnMount: &mountInfo,
+		},
+	}
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
Usage of `RetryOnConflict` doesn't seem to be following the usage instructions from the documentation. 
> RetryOnConflict is used to make an update to a resource when you have to worry about conflicts caused by other code making unrelated updates to the resource at the same time. fn should fetch the resource to be modified, make appropriate changes to it, try to update it, and return (unmodified) the error from the update function. On a successful update, RetryOnConflict will return nil. If the update function returns a "Conflict" error, RetryOnConflict will wait some amount of time as described by backoff, and then try again. On a non-"Conflict" error, or if it retries too many times and gives up, RetryOnConflict will return an error to the caller. 

---

What if the resource needs updated, not just created? What happens if multiple Rook operators with different versions are trying to control the resource.

Reply:
This was taken from the cluster-driver-registrar which was earlier doing this job https://github.com/kubernetes-csi/cluster-driver-registrar/blob/master/cmd/csi-cluster-driver-registrar/k8s_register.go#L67. updating is not required as of now.

Reply:
update with for loop instead of `RetryOnConflict`